randrctl/ctl.py

```diff
@@ -193,29 +194,80 @@ def _config_safe(self, config: dict, section: str, property: str):
+              config = os.path.join(homedir, config_format)
+              if os.path.isfile(config):
+                  configs[is_favourite].append(config)
+
+        # Return dictionary with the configuration files
+        return configs
+
+    def _load_config_files(self):
+        """
+        :return: dictionary of configurations loaded from config files
+        """
+        # Get config file paths
+        config_files = self._configs()
 
-    def get_randrctl(self):
-        config_files = list(self._configs())
-        config_files.reverse()  # reverse, so config file from the preferred home comes last (has highest precedence)
+        # If available, use preferred configs, otherwise use the others
+        use = 'preferred' if config_files.get('preferred') else 'other'
+
+        if use == 'other':
+          logger.warning("using system-wide configuration files")
+
+        # Fetch list of yaml files
+        yaml_files = [config_file for config_file in config_files[use] if config_file.endswith('yaml')]
+        # Fetch list of ini files
+        ini_files = [config_file for config_file in config_files[use] if config_file.endswith('ini')]
 
         # Empty container to hold the configuration
         config = {}
-        # Iterate over the list of config files
-        for config_file in config_files:
+
+        if yaml_files:
+          # Iterate over the list of config files
+          for config_file in yaml_files:
             with open(config_file, 'r') as stream:
-                try:
-                    # Try to parse the YAML file  and update the configuration dictionary
-                    config.update(yaml.load(stream))
-                except yaml.YAMLError as e:
-                    logger.error("error reading configuration file %s", config_file)
-                else:
-                    logger.debug("read configuration from %s", config_file)
+              # Try to parse the YAML file  and update the configuration dictionary
+              try:
+                config.update(load(stream))
+              except YAMLError as e:
+                logger.error("error reading configuration file %s", config_file)
+              else:
+                logger.debug("read configuration from %s", config_file)
+
+        elif ini_files:
+          logger.warning("loading INI configuration files")
+
+          # Instantiate config parser
+          config_parser = ConfigParser(allow_no_value=True, strict=False)
+
+          # Try to read the config files
+          try:
+            read = config_parser.read(ini_files)
+          except Exception as e:
+            logger.error("error reading configuration files")
+          else:
+            config = {section: dict(config_parser.items(section)) for section in config_parser.sections()}
+
+        # Otherwise, there are no configuration files whatsoever
+        else:
+          logger.error("there are no configuration files available")
```

Comment:
Maybe from this method point of view it is an error, but from application point of view it is not — _randrctl_ should work fine without configs present. We should not log this as error. As I proposed above, let's just log the name of the config file that we picked among all available